src/main/java/org/zalando/nakadi/service/subscription/StreamingContext.java

```diff
@@ -92,9 +92,20 @@ public long getKafkaPollTimeout() {
 
     @Override
     public void stream() throws InterruptedException {
+        // bugfix ARUHA-485
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> onShutdown()));
```

Comment:
I think we could add a unit test here by wrapping runtime in an object and inject it here. Then we could mock it and test that when we execute addShutdownHook it changes state in the expected manner.